libs/common/src/dtos/activity.dto.ts

```diff
@@ -0,0 +1,98 @@
+/**
+ * File name should always end with `.dto.ts` for swagger metadata generator to get picked up
+ */
+// eslint-disable-next-line no-shadow,max-classes-per-file
+export enum UnitTypeDto {
+  CM = 'cm',
+  M = 'm',
+  KM = 'km',
+  INCHES = 'inches',
+  FEET = 'feet',
+  MILES = 'miles',
+}
+
+// eslint-disable-next-line no-shadow
+export enum TagTypeDto {
+  Mention = 'mention',
+  Hashtag = 'hashtag',
+}
+
+// eslint-disable-next-line no-shadow
+export enum AttachmentTypeDto {
+  LINK = 'link',
+  IMAGE = 'image',
+  AUDIO = 'audio',
+  VIDEO = 'video',
+}
+
+export class LinkDto {
+  href: string;
+
+  name?: string;
+}
+
+export class LocationDto {
+  name: string;
+
+  accuracy?: number;
+
+  altitude?: number;
+
+  longitude?: number;
+
+  radius?: number;
+
+  units?: UnitTypeDto;
+}
+
+export class AssetReferenceDto {
+  referenceId: string;
+
+  height?: number;
+
+  width?: number;
+
+  duration?: string;
+}
+
+export class TagDto {
+  type: TagTypeDto;
+
+  name?: string;
+
+  mentionedId?: string;
+}
+
+export class AssetDto {
+  type: AttachmentTypeDto;
+
+  references: Array<AssetReferenceDto>;
+
+  name?: string;
+
+  href?: string;
+}
+
+export class BaseActivityDto {
+  name?: string;
+
+  tag?: Array<TagDto>;
+
+  location?: LocationDto;
+}
+
+export class NoteActivityDto extends BaseActivityDto {
```

Comment:
Side thought. Would we eventually want to add and export these out of`@dsnp/activity-content`?

Reply:
So the requests dto's didn't completely match the activity-contents implemented in that library. My first thought was to use those as DTOs but they have some extra params that we should be able to fill in ourselves. It made sense to create a separate and smaller DTO to simplify the fields that was needed to submit.

We would eventually need to use that library in workers when we are creating the Activity Contents from requuests.